index.js

```diff
@@ -396,6 +410,16 @@ function session(options) {
       });
     }
 
+    // check if session has been or is being destroyed
+    function isBeingDestroyed(req) {
+      return req.sessionID && req.session == null;
```

Comment:
Hmm, sounds like I must need another unit test for that last condition you mentioned. I'll take a peek when I get a chance.